cogs5e/models/sheet/coinpurse.py

```diff
@@ -0,0 +1,180 @@
+from cogs5e.models.errors import InvalidArgument
+from .mixins import HasIntegrationMixin
+from utils.functions import confirm
+
+CoinTypes = {
+    "pp": {
+        "icon": "<:DDBPlatinum:948681049326624849>",
+        "name": "Platinum",
+        "gSheet": {
+            "v14": "D72",
+            "v2": "D15",
+        }
+    },
+    "gp": {
+        "icon": "<:DDBGold:948681049221775370>",
+        "name": "Gold",
+        "gSheet": {
+            "v14": "D69",
+            "v2": "D12",
+        }
+    },
+    "ep": {
+        "icon": "<:DDBElectrum:948681048932364401>",
+        "name": "Electrum",
+        "gSheet": {
+            "v14": "D66",
+            "v2": "D9",
+        }
+    },
+    "sp": {
+        "icon": "<:DDBSilver:948681049288867930>",
+        "name": "Silver",
+        "gSheet": {
+            "v14": "D63",
+            "v2": "D6",
+        }
+    },
+    "cp": {
+        "icon": "<:DDBCopper:948681049217597480>",
+        "name": "Copper",
+        "gSheet": {
+            "v14": "D60",
+            "v2": "D3",
+        }
+    }
+}
+
+
+class Coinpurse(HasIntegrationMixin):
+    def __init__(self, pp=0, gp=0, ep=0, sp=0, cp=0):
+        super().__init__()
+        self.pp = pp
+        self.gp = gp
+        self.ep = ep
+        self.sp = sp
+        self.cp = cp
+
+    def __str__(self):
+        return "\n".join(self.str_styled(coin_type) for coin_type in CoinTypes)
+
+    def str_styled(self, coin_type):
+        if coin_type not in ("compact", "pp", "gp", "ep", "sp", "cp"):
+            raise ValueError("coin_type must be in ('compact', 'pp', 'gp', 'ep', 'sp', 'cp')")
+
+        if coin_type == 'compact':
+            coin_value = self.total
+            coin_type = 'gp'
+        else:
+            coin_value = getattr(self, coin_type)
+        return f"{CoinTypes[coin_type]['icon']} {coin_type}: {coin_value:,}"
```

Comment:
nit: I prefer `f"{CoinTypes[coin_type]['icon']} {coin_value:,} {coin_type}"` for the format of the coins; having the icon and the currency name right next to each other feels a little weird

Reply:
I was going for the field symmetry.   The coin textual representation in a straight line.  I like straight lines.  But, I do see that perhaps moving them around might be beneficial.   I can make the change and we can see how we feel when it's in use :)

Reply:
They'll still be in a straight line 😄
<img width="406" alt="Screenshot 2022-03-04 at 12 50 40" src="https://user-images.githubusercontent.com/8866981/156839693-94e57485-92d0-424a-a855-bfdbce8bce9c.png">